tests/posix/fs/src/test_fs_file.c

```diff
@@ -341,3 +341,16 @@ ZTEST(posix_fs_file_test, test_fs_fd_leak)
 		}
 	}
 }
+
+ZTEST(posix_fs_file_test, test_file_open_truncate)
+{
+	zassert_ok(test_file_open());
+	zassert_ok(test_file_write());
+	zassert_ok(test_file_close());
+	file = open(TEST_FILE, O_RDWR | O_TRUNC);
+	zassert_not_equal(file, -1,
+			  "File open failed for truncate mode");
+
+	zassert_ok(test_file_close());
+	zassert_ok(test_file_delete());
```

Comment:
Thanks for following up with this PR.

Prior to deleting the file, can you please also test that the file size is zero bytes?